meta-resin-common/recipes-support/resin-hostname/resin-hostname/resin-hostname

```diff
@@ -0,0 +1,19 @@
+#!/bin/sh
+
+set -e
+
+. /usr/sbin/resin-vars
+
+if [ -z "$NEW_HOSTNAME" ]; then
+    NEW_HOSTNAME=$(jq -r '.uuid //empty' $CONFIG_PATH)
+fi
+
+if [ -z "$NEW_HOSTNAME" ]; then
+    echo "[ERROR]: Unexpected uuid missing from config.json."
+    exit 1
+fi
+
+# Set hostname temporary hostname
+hostnamectl set-hostname "$NEW_HOSTNAME"
```

Comment:
Based on https://www.freedesktop.org/software/systemd/man/hostnamectl.html it seems we need /etc/machine-info too. Can you please address these in the ro PR?